cmd/kube-controller-manager/app/core.go

```diff
@@ -188,6 +189,24 @@ func startAttachDetachController(ctx ControllerContext) (bool, error) {
 	return true, nil
 }
 
+func startVolumeExpandController(ctx ControllerContext) (bool, error) {
+	if utilfeature.DefaultFeatureGate.Enabled(features.ExpandPersistentVolumes) {
+		expandController, expandControllerErr := expand.NewExpandController(
+			ctx.ClientBuilder.ClientOrDie("expand-controller"),
+			ctx.InformerFactory.Core().V1().PersistentVolumeClaims(),
+			ctx.InformerFactory.Core().V1().PersistentVolumes(),
+			ctx.Cloud,
+			ProbeExpandableVolumePlugins(ctx.Options.VolumeConfiguration))
```

Comment:
does this need a peer to `GetDynamicPluginProber(ctx.Options.VolumeConfiguration)`?

Reply:
We don't need `GetDynamicPluginProber` for this controller (at least for now), because currently only flex volumes are dynamically probed and they are not resizable.

Reply:
> currently only flex volumes are dynamically probed and they are not resizable.

why not? keeping parity in our out-of-tree volume mechanisms is important

Reply:
For resizing to work for Flex volumes, we will have to add expand/resize API calls to flex protocol. We are yet to be in a place where flex volumes can be called stable. Attach/Detach interface was added recently to flex and has issues that we are trying to workout - https://github.com/kubernetes/kubernetes/issues/44737. sig-storage is also working on making flex plugin deployments saner. 

In a nutshell, I would like to pursue flex resizing as separate problem. Several community members have expressed an interest in solving it. There is a separate feature request open for it- https://github.com/kubernetes/features/issues/304

Reply:
But just to clarify current resizing design does not inhibit a future flex resize design. As long as, protocols/API calls laid out by this design are implemented by flex - we will have working resizing. So it all comes down to implementing volume plugin interface necessary for resizing.

Reply:
understood. seems like that should tracked as part of graduating this to beta.

Reply:
ack. I will make a note of this in original design proposal.